src/classes/PMT_Payment_TDTM.cls

```diff
@@ -132,6 +157,55 @@ public class PMT_Payment_TDTM extends TDTM_Runnable {
         // the Opportunity should still be marked as closed.
         PMT_PaymentCreator pc = new PMT_PaymentCreator(mapIdOppNew, mapIdOppOld, triggerAction, dmlWrapper);
 
+       
+        // Prevent the user from modifying the Opp Amount
+        if (triggerAction == TDTM_Runnable.Action.AfterUpdate && isEnforceAccountingDataConsistency) {
+            Set<Id> oppIds = (new Map<Id, Opportunity>(newList)).keySet();
+            Map<Id, Opportunity> oppsById = getPaymentOpportunities(oppIds);
+
+            for (Opportunity opp : newList) {
+                // get the opp with payments
+                Opportunity oppWithPayment = oppsById.get(opp.Id);
+
+                Decimal totalPaid = 0;
```

Comment:
Calculating the total Paid payment amount could be a shared method in the OPP_OpportunityAmountSync_SVC class that it and this class use.(170 - 173)